src/tests/java/htsjdk/samtools/CRAMIndexerTest.java

```diff
@@ -0,0 +1,70 @@
+package htsjdk.samtools;
+
+import htsjdk.samtools.cram.build.ContainerFactory;
+import htsjdk.samtools.cram.structure.Container;
+import htsjdk.samtools.cram.structure.CramCompressionRecord;
+import htsjdk.samtools.cram.structure.Slice;
+import htsjdk.samtools.seekablestream.SeekableMemoryStream;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Created by vadim on 12/01/2016.
+ */
+public class CRAMIndexerTest {
+
+    private static CramCompressionRecord createRecord(int recordIndex, int seqId, int start) {
+        byte[] bases = "AAAAA".getBytes();
+        int readLength = bases.length;
+
+        final CramCompressionRecord record = new CramCompressionRecord();
+        record.setSegmentUnmapped(false);
+        record.setMultiFragment(false);
+        record.sequenceId = seqId;
+        record.alignmentStart =start;
+        record.readBases = record.qualityScores = bases;
+        record.readName = Integer.toString(recordIndex);
+        record.readLength = readLength;
+        record.readFeatures = Collections.emptyList();
+
+        return record;
+    }
+    @Test
+    public void test_processMultiContainer() throws IOException, IllegalAccessException {
```

Comment:
Also, since this is a new file, can we rename it to something that reflects that its BAI index oriented, i.e., CRAMBAIIndexerTest or something like that.